electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -142,6 +142,15 @@ def _get_unique_source_type(cls, events: pd.DataFrame) -> EventSourceType:
 class ExchangeList(AggregatableEventList):
     events: list[Exchange]
 
+    def __contains__(self, datetime) -> bool:
+        return any(event.datetime == datetime for event in self.events)
+
+    def __getitem__(self, datetime) -> Exchange:
+        return next(event for event in self.events if event.datetime == datetime)
+
+    def __setitem__(self, datetime, event: Exchange):
+        self.events[self.events.index(self[datetime])] = event
```

Comment:
These can potentially be moved up to the parent classes and be re-used between all the parser class lists if we want to.

Reply:
Those changes make sense to me, but I think they should be in the parent class rather.

Reply:
That is what I do in the follow up PR, but I can do it directly in here as well and then update the other PR to just contain the logic changes I want to make.